eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -121,7 +126,14 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
       myStop()
 
     case Event(fail: UpdateFailHtlc, WaitingForComplete(s, c, _, failures, sharedSecrets, ignoreNodes, ignoreChannels, hops)) =>
-      Sphinx.FailurePacket.decrypt(fail.reason, sharedSecrets) match {
+      val tryDecrypt = Sphinx.FailurePacket.decrypt(fail.reason, sharedSecrets)
+      tryDecrypt match {
+        case Success(e) =>
+          Metrics.PaymentError.withTag(Tags.Failure, Tags.FailureType.get(RemoteFailure(Nil, e))).increment()
+        case Failure(_) =>
+          Metrics.PaymentError.withTag(Tags.Failure, Tags.FailureType.get(UnreadableRemoteFailure(Nil))).increment()
+      }
+      tryDecrypt match {
```

Comment:
Without intermediate variable (and because I wanted to try multi-line suggestions):

```suggestion
      (Sphinx.FailurePacket.decrypt(fail.reason, sharedSecrets) match {
        case s@Success(e) =>
          Metrics.PaymentError.withTag(Tags.Failure, Tags.FailureType.get(RemoteFailure(Nil, e))).increment()
          s
        case f@Failure(_) =>
          Metrics.PaymentError.withTag(Tags.Failure, Tags.FailureType.get(UnreadableRemoteFailure(Nil))).increment()
          f
      }) match {
```